test/integration_tests.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:treasure_of_the_high_seas/model/audio/audio_constants.dart';
+import 'package:treasure_of_the_high_seas/model/settings/settings_model.dart';
+
+import 'mocks.dart';
+import 'model/audio/audio_model_test.dart';
+import 'test_utils.dart';
+
+void main() {
+
+  testWidgets('should start playing game music when music setting toggled on from in game', (WidgetTester tester) async {
+    SharedPreferences.setMockInitialValues(
+        {AppSetting.musicEnabled.toString(): false});
+
+    final musicCache = MockAudioCache();
+    final settingsModel = SettingsModel(await SharedPreferences.getInstance());
+    var audioModel = await makeAudioModel(settingsModel: settingsModel, musicCache: musicCache);
+
+    await launchGameFromMenu(tester, audioModel: audioModel);
+
+    settingsModel.updateSetting(AppSetting.musicEnabled, true);
```

Comment:
If we really wanted to go for it with the integration test, we could go further here and do something like:

```
    final settingsFinder = find.byIcon(Icons.settings);
    await tester.tap(settingsFinder);
    await tester.pumpAndSettle();

    final musicSwitchFinder = find.widgetWithText(SwitchListTile, 'Music');
    await tester.tap(musicSwitchFinder);
    await tester.pumpAndSettle();
```

For this to work, we'd need to tweak the `launchGameFromMenu` and `launchWidget` helpers to take an optional `settingsModel` - so `launchWidget` becomes:

```
Future<void> launchWidget({Widget child, GameState state, AudioModel audioModel, SettingsModel settingsModel}) async {
    final prefs = await runAsync(() async => await SharedPreferences.getInstance());

    final app = MultiProvider(providers: [
      ChangeNotifierProvider<GameState>.value(value: state),
      ChangeNotifierProvider<SettingsModel>.value(value: settingsModel ?? SettingsModel(prefs)),
      Provider<AudioModel>.value(value: audioModel ?? MockAudioModel())
    ], child: MaterialApp(home: child));

    await pumpWidget(app);
    await pumpAndSettle();
  }
```